app/controllers/api_keys_controller.rb

```diff
@@ -15,7 +15,8 @@ def new
 
   def create
     key = generate_unique_rubygems_key
-    @api_key = current_user.api_keys.build(api_key_params.merge(hashed_key: hashed_key(key)))
+    build_params = { user: current_user, hashed_key: hashed_key(key), **api_key_params }
+    @api_key = ApiKey.new(build_params)
```

Comment:
setting `rubygem_id` is dependant on having user available, calling `build` on current_user doesn't set the user until the rest of the params have been set.